network-store-client/src/main/java/com/powsybl/network/store/client/AbstractTopology.java

```diff
@@ -0,0 +1,292 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.network.store.client;
+
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.ConnectableType;
+import com.powsybl.iidm.network.ThreeWindingsTransformer;
+import com.powsybl.network.store.model.*;
+import org.jgrapht.UndirectedGraph;
+import org.jgrapht.alg.ConnectivityInspector;
+import org.jgrapht.graph.Pseudograph;
+
+import java.util.*;
+import java.util.function.BiPredicate;
+import java.util.stream.Collectors;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public abstract class AbstractTopology<T> {
+
+    private final BiPredicate<Map<T, List<Vertex<T>>>, Set<T>> busViewBusValidator = (verticesByNodeOrBus, nodesOrBuses) -> {
```

Comment:
replace with     private static <E> boolean busViewBusValidator (Map<E, List<Vertex<E>>>verticesByNodeOrBus, Set<E> nodesOrBuses) {
// update the types from E to T in body
//not sure if it's better to change the name from E to T or if we should just shadow the T of the class..
}

and use method reference AbstractTopology::busViewBusValidator at the bottom of the class

Reply:
yeah maybe just shadow T

Reply:
Done